ASM/c/dungeon_info.c

```diff
@@ -293,12 +295,30 @@ void draw_dungeon_info(z64_disp_buf_t* db) {
 
         left += icon_size + padding;
 
-        // Draw dungeon names
-
+        // Draw the list of dungeons.
+        // Pre completed dungeons are grayed and crossed out.
         for (int i = 0; i < rows; i++) {
+            gDPPipeSync(db->p++);
             dungeon_entry_t* d = &(dungeons[i]);
+            int empty = CFG_DUNGEON_PRECOMPLETED[d->index];
             int top = start_top + ((icon_size + padding) * i) + 1;
-            text_print_size(db, d->name, left, top, font_width, font_height);
+            if (empty == 1) {
+                gDPSetPrimColor(db->p++, 0, 0, 0xFF, 0xFF, 0xFF, 0x7F);
+                int sizeRectangle = text_print_size(db, d->name, left, top, font_width, font_height) - left;
```

Comment:
Can this be guaranteed to never be negative? If so, it should be unsigned. This will allow the compiler to optimize to bit shifting for the multiplication by a power of two if you make the changes I requested there. Maybe we should have a MAX macro to ensure that this is above 0 before assigning it to enforce the guarantee.

Reply:
Similarly I think the `left` variable should be inspected to see if it can be made unsigned and guaranteed to be 0 or larger as it is used more in the below calculations.

This is all assuming you were doing the manual optimization on purpose though.

Reply:
text_print_size's minimal return value is left so looks good to go to unsigned indeed